### DISCUSSION TITLE
Know which connectors are available

### DISCUSSION THREAD

**Discussion by cmi23** [wevm/wagmi#1038] [Q&A]
Is there a way of knowing if a connector is available before calling `connect`?

I'm working on an app that allows the user to connect with a wallet of their choice (metamask or coinbase), but not everyone has both installed. Does `wagmi` provide a way of detecting which connectors can be used or which wallets the user has installed?

I tried looking at the `connectors` field but it looks like it only returns one connector even if I have 2 wallets installed.

**Comment by tmm** [comment]
Check out `Connector`'s `ready` property. Below, we disable an option if they wallet is not available.

```tsx
import { useConnect } from 'wagmi'

function App() {
  const { connect, connectors, error, isLoading, pendingConnector } =
    useConnect()

  return (
    <div>
      {connectors.map((connector) => (
        <button
          disabled={!connector.ready}
          key={connector.id}
          onClick={() => connect({ connector })}
        >
          {connector.name}
        </button>
      ))}

      {error && <div>{error.message}</div>}
    </div>
  )
}
```
